internal/pkg/agent/application/info/agent_metadata.go

```diff
@@ -151,7 +152,10 @@ func (i *AgentInfo) ECSMetadata(l *logger.Logger) (*ECSMeta, error) {
 	info := sysInfo.Info()
 	hostname := info.Hostname
 	if features.FQDN() {
-		fqdn, err := sysInfo.FQDN()
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
```

Comment:
isn't all this (L152-L164) what you have in `getHostName`?

Reply:
Yes, good spot.  I need to move that function to a common location so I can share it between two two packages.

Reply:
Refactored in f4878696dc. Thanks for catching.

Reply:
Refactored in f4878696dc. Thanks for catching.

Reply:
Added unit tests as well: f2f3355cc2